src/anndata/experimental/_dispatch_io.py

```diff
@@ -42,6 +43,32 @@ def read_dispatched(
     return reader.read_elem(elem)
 
 
+async def read_dispatched_async(
```

Comment:
Maybe this doesn’t need to be a separate function.

We could just check if the callback returns something awaitable, right?

Reply:
So we would have just 

```python
def read_dispatched (...) -> Coroutine | AnnData
````

and then it would be up to users to deal with it?  Or you mean that all of the "duplicated" code should be made to just handle coroutines and not (which I kind of like)